peer/peer.go

```diff
@@ -150,25 +215,48 @@ func ExtractPeerInfo(peerma string) (lpeer.ID, ma.Multiaddr, error) {
 	return peerid, trgtAddr, nil
 }
 
-// Connect adds the given multiaddress to p's Peerstore and opens a stream
-// with the peer at that multiaddress if the multiaddress is valid, otherwise
-// returns error.
-func (p *Peer) Connect(peerma string) (net.Stream, error) {
-	peerid, targetAddr, err := ExtractPeerInfo(peerma)
-	if err != nil {
-		return nil, err
+// advertisePeers writes messages into the given stream advertising the
+// multiaddress of each peer in this peer's subnet.
+func (p *Peer) advertisePeers(s net.Stream) {
```

Comment:
Why do we need to advertise peers? Why not just give our list of peers during connection, or accept requests for our list of peers. Seems like an an unnecessary complication